src/rpy_diffusive_flux.cc

```diff
@@ -33,9 +35,30 @@ void RPYDiffusiveFlux::compute(std::shared_ptr<GrandPotential> grand, std::share
         {
         throw std::invalid_argument("Spherical geometry required");
         }
-
+    
+    if(state->getNumFields() > 1)
+        {
+        throw std::invalid_argument("Only one type of particle is supported");
+        }
+        
+    const GridInterpolator& g = *mobility_;
+    
+    const double max_x = std::get<0>(g.getUpperBounds());
+    const double cutoff = std::get<1>(g.getUpperBounds());
+    
+    if(std::get<1>(g.getLowerBounds()) != -cutoff)
+        {
+        throw std::invalid_argument("Cutoff bounds for integration are not valid");
+        }
+    
+    const double max_density = std::get<2>(g.getUpperBounds());
```

Comment:
For readability, I would read this at the same time as the other things, then do the error check
```suggestion
    const double max_density = std::get<2>(g.getUpperBounds());

    if(std::get<1>(g.getLowerBounds()) != -cutoff)
        {
        throw std::invalid_argument("Cutoff bounds for integration are not valid");
        }

```